rosbag2_compression/src/rosbag2_compression/sequential_compression_writer.cpp

```diff
@@ -205,6 +205,7 @@ void SequentialCompressionWriter::compress_last_file()
   if (!compressor_) {
     throw std::runtime_error{"Compressor was not opened!"};
   }
+  storage_.reset();  // Storage should be closed before compression
```

Comment:
Why?

Reply:
Doesn't it make more sense to perform the reset in ```split_bagfile``` since that's where reassignment happens?

Reply:
If you move it to `split_bagfile`, be sure to mirror it in `reset`

Reply:
Update: Added this into `compression-cli` PR and it passed on a local windows run.